### ISSUE TITLE
Ability to specify offset day for financial year start

### ISSUE THREAD

**Issue by FreneticScribbler** [InvoiceShelf/InvoiceShelf#191]
As per https://github.com/crater-invoice/crater/issues/34 on Crater, one of the major pain points of this platform for me is still that I have to use 'custom' date ranges to get an accurate picture of my P+L etc because the UK financial year runs from the 6th of April.

Not sure if other countries are the same - the quick and dirty option might be to have a 'UK' option that hardcodes our FY - definetely very reasonable IMHO if we're the only country with a non month aligned FY... 

I'd love to open a PR and implement this myself but whilst I normally try and only host/use software written in languages I can work in, this is an unfortunate exception. I may instead be able to put a bounty on developing this feature given it'd save me quite a notable amount of clicks/time, if that's possible?

**Comment by aybanda** [comment]
Hi @FreneticScribble

I'd like to volunteer to tackle this issue. I understand the importance of aligning with the UK fiscal year and the frustration of constantly adjusting date ranges.

Here's a concise plan to implement this feature:

1. Add fiscal year settings to user preferences
   - Custom start date (day/month)
   - UK preset (April 6th)
2. Implement fiscal year calculation logic
3. Update reports/dashboards to use custom fiscal year
4. Ensure backward compatibility
5. Add unit tests
6. Update documentation

Estimated time: 3 days

@FreneticScribbler By the way, could you clarify the details of the bounty you mentioned? It might help in prioritizing this task.

Looking forward to contributing to this.

**Comment by FreneticScribbler** [comment]
To be honest I'm not even sure that's the right word, I just figured that fair is fair given I use the software as part of my business which makes me money (sometimes!) and am unable to contribute the time myself. I don't really have any idea how it would work in practice, though..

**Comment by rihards-simanovics** [comment]
@FreneticScribbler careful, I smell ChatGPT in the comment from @aybanda. If you do decide to work together make sure to have a contract in place so your money don't valis in thin air. if you wish to donate, I'd suggest speaking with @gdarko as he is the main maintainer for this project.

**Comment by aybanda** [comment]
Hey @rihards-simanovics 

Just to clear things up - I'm not ChatGPT or any other AI. I'm a real person who got a bit too excited about contributing and ended up sounding like a corporate email. My bad!

@FreneticScribbler and @rihards-simanovics, sorry if I came across as weird or fake. I promise I'm just a dev who loves solving problems and got carried away trying to sound professional.

@gdarko, I'd love to hear your thoughts on this whole situation. If you've got a process for checking out new contributors, I'm totally up for it. Happy to show some of my past work or chat about my experience if that helps.

**Comment by aybanda** [comment]
Also @rihards-simanovics, do we have any established channels for users who want to financially support feature development? If not, this might be a good opportunity to consider setting something up.

**Comment by rihards-simanovics** [comment]
no worries, I'm just being overly cautious due to bounty being mentioned. Thanks for clarifying though. Hey @gdarko never really done this before but is there any platforms that allow people to post bounties for features and the one who completes the feature gets the reward?

**Comment by aybanda** [comment]
Yes there are plenty bounty platforms since I've been poking a lot on github, I have found few  platforms that work just fine.
You could consider exploring them for your feature I will share the details with you soon.

For this issue,I've got an idea to throw out here. How about we keep things simple?
@gdarko, @rihards-simanovics, what if you assign this UK tax year feature to me? I'll dive in, get it done, and once it's all wrapped up and you're happy with how it works, we can chat about payment then.

Just to give a ballpark figure, based on the work involved (about 3 days), we're looking at somewhere between $1,200 to $2,400. But that's flexible - we can figure out the details after you've seen the feature in action and are satisfied with it.

@FreneticScribbler, this way you don't have to worry about any bounty platforms or anything complicated. (at least for this issue) 

If the feature turns out great and solves your problem, we can sort out the contribution then.
What do you all think? This way we can focus on getting the feature built first, and deal with the money stuff later.

**Comment by rihards-simanovics** [comment]
>@gdarko, @rihards-simanovics, what if you assign this UK tax year feature to me? I'll dive in, get it done, and once it's all wrapped up and you're happy with how it works, we can chat about payment then.

> Just to give a ballpark figure, based on the work involved (about 3 days), we're looking at somewhere between $1,200 to $2,400. But that's flexible - we can figure out the details after you've seen the feature in action and are satisfied with it.

As this is an open-source project, we primarily rely on user contributions. Currently, the project is not funded in a way that would allow us to pay for feature creation, let alone the time we ourselves contribute. Security issues might be an exception, but this is at the discretion of Darko.

User contributions in the form of pull requests are always welcome. However, at this time, it is up to the original poster of the feature request to provide a bounty, exchange contacts, and negotiate the contract with the feature developer.

All future discussions about the bounty in this issue will be marked as `of topic`.

**Comment by FreneticScribbler** [comment]
Apologies. Shouldn't have dropped that offhand like that without thinking of what a can of worms I was opening.

**Comment by rihards-simanovics** [comment]
> Apologies. Shouldn't have dropped that offhand like that without thinking of what a can of worms I was opening.

Please @FreneticScribbler, no need to apologise, I'm sure I'll speak on behalf of the Invoice Shelf team when I say that it is always appreciated when the community is willing to support the project going as far as creating a bounty for a feature, they wish to be implemented 😊.

If you wish to pursue the bounty as I said from the beginning feel free to exchange the details with @FreneticScribbler, you may find it useful to open a discussion for this one in particular [here](https://github.com/InvoiceShelf/InvoiceShelf/discussions). I would like to keep issues clear of any off-topic discussions i.e. discussions unrelated to the issues fixes or genuine help.

**Comment by rihards-simanovics** [comment]
@gdarko I think I found a bounty program that allows users to put a bounty on the feature request: https://console.algora.io/. We can also integrate this into GitHub issues, which is how I found it.

It was implemented on the backups feature request issue of Coolify: https://github.com/coollabsio/coolify/issues/2389
